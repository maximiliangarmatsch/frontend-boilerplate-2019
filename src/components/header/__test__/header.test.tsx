import React from 'react';
import { render, shallow } from 'enzyme';
// Template Component
import Header from '../header.component';

describe('<Header />', () => {
    test('renders', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
    test('header should have links', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('Link').length).toEqual(1);
    });
    test('links should have props', () => {
        const wrapper = shallow(<Header />);
        wrapper.find('Link').forEach(element => {
            expect(typeof element.at(0).prop('to')).toBe('string');
            expect(typeof element.at(0).text()).toBe('string');
        });

    });
});