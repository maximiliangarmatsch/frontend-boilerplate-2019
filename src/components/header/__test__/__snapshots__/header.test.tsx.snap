// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<header>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(AppBar))
      position="static"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-3"
          variant="h6"
        >
          <Link
            className="makeStyles-menuButton-2"
            key="1"
            to="/about"
          >
            About
          </Link>
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
  </div>
</header>
`;
